facebook/screenshot-tests-for-android
# Make accessibility info for the screenshot an optional value

### :pushpin: References
* **Issue:** Closes #248 
* **Related PRs:** #249

### :tophat: What is the goal?

Make the usage of the accessibility report feature optional and configurable from the public API.

**In order to properly understand the motivations behind this PR and get full context we'd recommend you reading issue #248 and the PR #249.**

### How is it being implemented?

We've just updated and documented the public ``RecordBuilder`` API and ``RecourdBuilderImpl`` implementation to be able to handle a new parameter related to the usage of the accessibility feature. Once this was ready, we've modified the record process to do not include the accessibility information if this configuration flag is disabled. **In order to do not introduce any breaking change the parameter default value enables this feature.** Last but not least, I've added end to end coverage in the sample project ensuring the library works with this optional parameter disabled.

### How can it be tested?

I've created a new test in the sample project disabling this flag to ensure the library is still working in an end to end scenario. However, I don't see any CI tool configured checking the codestyle of the code or evaluating these tests so I can only ensure this worked in my machine 😃 

### Important considerations and questions.

We already agree this is not a fix for the original issue but a workaround to be able to use the library with views where the accessibility information generation is not working. As the error is placed in a library we don't have access to and there is no stable version fixing the exception throw in the original library we think this would be a nice workaround for this feature.
